r/tests/testthat/test-compute-sort.R

```diff
@@ -0,0 +1,157 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+context("compute: sorting")
+
+library(dplyr)
+
+tbl <- example_data_for_sorting
+
+test_that("Scalar sort", {
+  expect_identical(
+    as.vector(sort(Scalar$create(42L))),
+    42L
+  )
+})
+
+test_that("Array sort on integers", {
+  expect_equal(
+    Array$create(tbl$int)$SortIndices(),
+    Array$create(0L:9L, type = uint64())
+  )
+  expect_equal(
+    Array$create(rev(tbl$int))$SortIndices(descending = TRUE),
+    Array$create(c(1L:9L, 0L), type = uint64())
+  )
+  expect_equal(
```

Comment:
I believe there is an `expect_vector()` (defined in `helper-expectations.R`) that can help with these

Reply:
I added a new helper function in I added a new function `expect_vector_equal()` in ddaffcd41b899a79f95f08717544db3b3266d273. It works like `expect_dplyr_equal()`.